src/g3log/logmessage.hpp

```diff
@@ -87,7 +88,8 @@ namespace g3 {
       // Complete access to the raw data in case the helper functions above
       // are not enough.
       //
-      std::time_t _timestamp;
+      // TODO: unique_ptr should be considered instead of shared_ptr
+      std::shared_ptr<Timestamp> _timestamp;
```

Comment:
I do not see the reason for having a unique or shared pointer for this. By introducing a shared_ptr you will also introduce quite a bit of overhead and possibility of cache misses.   

Since the pull request targets time sensitive applications I do not think this is in your interest. _(it's not in mine)_